### PR TITLE
minor fix to as.ITime

### PR DESCRIPTION
**Pull request by MichaelChirico**
This comment confuses me:

```
  if (missing(date) && inherits(tz, c("Date", "IDate", "POSIXt", "dates"))) {
    date <- tz # allows you to use date as the 2nd argument
    tz <- "UTC"
  }
```

Why not just leave `date` as the second argument? I usually write `tz = ` in code whenever I specify it anyway...

---

Also noticed inconsistent usage of `...` vs `\dots` in the manual.... updating...

### CODE REVIEW

**Comment by MichaelChirico** [comment]
@jangorecki I figured there might be a reason, but none came to mind. So, I raised this PR 😎 

`?Sys.Date` isn't too clear to me:

> `Sys.time` and `Sys.Date` returns the system's idea of the current date with and without time.

Strongly implies to me that one is just a truncation of the other (i.e. `Sys.Date()` is fine as default); but further down

> `Sys.time` returns an absolute date-time value which can be converted to various time zones and may return different days.

> `Sys.Date` returns the current day in the current time zone.

Which is more ambiguous to me. FWIW I just spoofed my machine's time to one where there's a different date in local vs. UTC timezone, and the former theory seems to hold up (i.e. `Sys.Date()` OK)

```
Sys.Date()
[1] "2018-10-03"
> Sys.time()
[1] "2018-10-03 00:05:14 +08"
```

**Comment by MichaelChirico** [comment]
Also some minor confusion about recommended practice for `\dots` from the [WRE manual]()

> There is only a fine distinction between `\dots` and `\ldots`. It is technically incorrect to use `\ldots` in code blocks and `tools::checkRd` will warn about this—on the other hand the current converters treat them the same way in code blocks, and elsewhere apart from the small distinction between the two in LaTeX.

Following this, I used `\dots` wherever `...` appeared in a region where the output will be preformatted style. But this suggests it doesn't need to be as strict (can restrict to _only_ function argument lists):

> Use `\dots` for the dots in function argument lists `…`, and `\ldots` for ellipsis dots in ordinary text.

**Comment by MichaelChirico** [comment]
@jangorecki Assuming we go with removing all verbatim `...` from the manual, how easy/recommended would it be to add something like `grep -Fq ... man/*` to the CI?

File: man/chmatch.Rd [hunk 1]
```diff
@@ -48,7 +48,7 @@ N = 1e5
 u = as.character(as.hexmode(1:10000))
 y = sample(u,N,replace=TRUE)
 x = sample(u)
-                                           #  With N=1e8 ...
+                                           #  With N=1e8 \dots
```
Comment (jangorecki) [inline man/chmatch.Rd]: like here

File: man/fread.Rd [hunk 2]
```diff
@@ -174,7 +174,7 @@ if(all(sapply(c("sqldf", "ff"), requireNamespace, quietly = TRUE))) {
   identical(DF1,within(as.data.frame(FFDF),d<-as.character(d)))
 }
 
-# Scaling up ...
```
Comment (jangorecki) [inline man/fread.Rd]: and here

**Comment by MichaelChirico** [comment]
NVM, `\dots` in `\usage` section renders as-is anyway, so it's a moot point. Will revert those...

**Comment by mattdowle** [comment]
```R
> Sys.Date
function () 
as.Date(as.POSIXlt(Sys.time()))
```

That `as.POSIXlt` is appallingly slow.  In this context it would affect call overhead. `POSIXlt` should never be used by data.table, directly or indirectly. It uses 40 bytes to store a date!  If I had a say I would purge `POSIXlt` from R.

**Comment by mattdowle** [comment]
We can do this dots thing,  but is it really causing any problems?   We have 700 open issues which are higher priority than that.

**Comment by MichaelChirico** [comment]
Good call on the `Sys.Date` source... can't believe I never noticed that before! Indeed half as fast, but on microsecond scale...

```
microbenchmark(times = 1e4,
               as.Date(as.POSIXlt(Sys.time())),
               as.Date(Sys.time()))
# Unit: microseconds
#                             expr    min     lq     mean  median
#  as.Date(as.POSIXlt(Sys.time())) 14.935 17.550 20.73792 18.2250
#              as.Date(Sys.time())  6.022  7.664 10.04162  8.3955
#      uq       max neval
#  19.699  9874.823 10000
#   9.173 10126.756 10000
```

If call overhead is the concern, why not default to have the `%F` portion of `as.POSIXct.ITime` default to `1970-01-01`? Why not just set the default to `date = structure(0L, class = 'Date')`... could even write a utility function `.Date = function(x) structure(x, class = 'Date')` (mirroring the `.POSIXct` function) to make it prettier?

FWIW, `as.POSIXlt` is used in the body of the function as well (don't think it can be avoided):

```
as.POSIXct(as.POSIXlt(date), tz = tz) + x
```

Regarding `\dots`, I agree, but the fix is already done :)

**Comment by mattdowle** [comment]
Ok. Good benchmark.

File: man/data.table.Rd [hunk 3]
```diff
@@ -61,7 +61,7 @@ data.table(..., keep.rownames=FALSE, check.names=FALSE, key=NULL, stringsAsFacto
 
             \item{If not, \code{x} \emph{must be keyed}. Key can be set using \code{\link{setkey}}. If \code{i} is also keyed, then first \emph{key} column of \code{i} is matched against first \emph{key} column of \code{x}, second against second, etc..
 
-            If \code{i} is not keyed, then first column of \code{i} is matched against first \emph{key} column of \code{x}, second column of \code{i} against second \emph{key} column of \code{x}, etc...
+            If \code{i} is not keyed, then first column of \code{i} is matched against first \emph{key} column of \code{x}, second column of \code{i} against second \emph{key} column of \code{x}, etc\ldots
```
Comment (jangorecki) [inline man/data.table.Rd]: @MichaelChirico another unexpected `\dots`

**Pull request merged**